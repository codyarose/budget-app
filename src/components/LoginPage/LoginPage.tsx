import React, { FC, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { loginUser, selectIsAuthed } from '../../features/auth/authSlice'
import { RouteComponentProps } from 'react-router-dom'

type Props = RouteComponentProps

const LoginPage: FC<Props> = ({ history }) => {
	const dispatch = useDispatch()
	const isAuthed = useSelector(selectIsAuthed)

	const handleClick = () => {
		dispatch(loginUser())
	}

	useEffect(() => {
		isAuthed && history.push('/')
	}, [isAuthed, history])

	return (
		<div>
			<h3>Login</h3>
			<button onClick={handleClick}>Login</button>
		</div>
	)
}

export default LoginPage
