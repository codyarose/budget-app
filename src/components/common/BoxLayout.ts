import styled from 'styled-components'

const Layout = styled.div`
	background: ${({ theme }) => theme.colors.white};
	height: 100vh;
	display: flex;
	justify-content: center;
	align-items: center;
`

const Box = styled.div`
	position: relative;
	background-color: ${({ theme }) => theme.colors.white};
	border: 3px solid ${({ theme }) => theme.colors.black};
	text-align: center;
	padding: ${({ theme }) => `${theme.spacing.md} ${theme.spacing.lg}`};
	box-shadow: ${({ theme }) =>
		`${theme.spacing.xs} ${theme.spacing.xs} 0 currentColor`};
`

const BoxLayout = {
	Layout,
	Box,
}

export default BoxLayout
