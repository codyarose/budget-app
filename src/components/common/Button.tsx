import React, { FC } from 'react'
import styled, { css } from 'styled-components'
import { transitions } from 'polished'

interface Props {
	onClick?: () => void
	className?: string
	title?: string
}

const Button: FC<Props> = ({ onClick, children, className, title }) => {
	return (
		<StyledButton onClick={onClick} className={className} title={title}>
			{children}
		</StyledButton>
	)
}

export default Button

export const buttonStyles = css`
	background-color: ${({ theme }) => theme.colors.bg};
	color: ${({ theme }) => theme.colors.fg};
	border: 2px solid currentColor;
	padding: ${({ theme }) => `${theme.spacing.xs} ${theme.spacing.sm}`};
	font-size: 1rem;
	box-shadow: 3px 3px 0 currentColor;
	cursor: pointer;
	${transitions(
		'box-shadow 0.1s ease-in-out',
		'transform 0.1s ease-in-out',
		'background-color 0.2s ease-in-out',
		'color 0.2s ease-in-out',
	)}
	&:hover,
	&:focus {
		box-shadow: 5px 5px 0 currentColor;
		outline: none;
	}
	&:active {
		transform: translate(1px, 1px);
		box-shadow: 4px 4px 0 currentColor;
	}
`

const StyledButton = styled.button`
	${buttonStyles}
`
