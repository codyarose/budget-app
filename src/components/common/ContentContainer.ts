import styled from 'styled-components'

const ContentContainer = styled.div<{ padBottom?: boolean }>`
	width: 100%;
	max-width: 64rem;
	margin: 0 auto;
	padding: ${({ theme }) => `0 ${theme.spacing.md}`};
	${(props) =>
		props.padBottom && `padding-bottom: ${props.theme.spacing.lg};`}
	@media screen and (max-width: 450px) {
		padding: ${({ theme }) => `0 ${theme.spacing.sm}`};
	}
`

export default ContentContainer
