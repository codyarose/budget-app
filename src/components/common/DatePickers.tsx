import React, { FC } from 'react'
import styled, { css } from 'styled-components'
import { rgba } from 'polished'
import {
	DateRangePicker,
	DateRangePickerShape,
	SingleDatePicker,
	SingleDatePickerShape,
} from 'react-dates'

export const StyledDateRangePicker: FC<DateRangePickerShape> = (props) => {
	return (
		<DateRangeWrapper>
			<DateRangePicker {...props} />
		</DateRangeWrapper>
	)
}

export const StyledSingleDatePicker: FC<SingleDatePickerShape> = (props) => {
	return (
		<SingleDateWrapper>
			<SingleDatePicker {...props} />
		</SingleDateWrapper>
	)
}
StyledSingleDatePicker.displayName = 'StyledSingleDatePicker'

const commonStyles = css`
	--background: ${({ theme }) => theme.colors.bg};
	--foreground: ${({ theme }) => theme.colors.fg};
	color: var(--foreground);
	transition: color 0.2s ease-in-out;

	.DayPicker__horizontal,
	.CalendarMonth,
	.CalendarMonthGrid {
		background: inherit;
	}
	.DateInput_input,
	.DayPicker_weekHeader,
	.CalendarMonth_caption {
		color: inherit;
	}
	.CalendarDay__default,
	.DayPickerNavigation_button__default {
		background: inherit;
		color: inherit;
	}

	.DateInput {
		height: 100%;
		background-color: transparent;
	}
	.DateInput_input {
		height: 100%;
		background-color: inherit;
		&__focused {
			border-color: var(--foreground);
		}
	}
	.DayPicker {
		&__withBorder {
			box-shadow: ${({ theme }) =>
				`0 2px 6px ${rgba(theme.colors.fg, 0.05)}, 0 0 0 1px ${rgba(
					theme.colors.fg,
					0.07,
				)}`};
		}
		&__horizontal {
		}
		&_weekHeader {
		}
	}
	.CalendarMonth,
	.CalendarMonthGrid {
	}
	.CalendarMonth_caption {
	}
	.CalendarDay {
		&__default {
		}
		&__selected {
			background: var(--foreground);
			border-color: var(--foreground);
			color: var(--background);
		}
		&__selected_span {
			background: ${({ theme }) => rgba(theme.colors.fg, 0.5)};
			border-color: ${({ theme }) => rgba(theme.colors.fg, 0.25)};
			color: var(--background);
			&:hover {
				background: ${({ theme }) => rgba(theme.colors.fg, 0.6)};
			}
		}
	}
	.DayPickerKeyboardShortcuts_show__bottomRight::before {
		border-right-color: var(--foreground);
	}
	.DayPickerKeyboardShortcuts_showSpan {
		color: var(--background);
	}
	.DayPickerNavigation_button__default {
		border: 1px solid ${({ theme }) => rgba(theme.colors.fg, 0.5)};
	}
	.DayPickerNavigation_svg__horizontal {
		fill: var(--foreground);
	}
`

const SingleDateWrapper = styled.div`
	.SingleDatePicker {
		width: 100%;
	}
	.SingleDatePickerInput {
		width: 100%;
		background-color: transparent;

		&__withBorder {
			border: none;
			border-bottom: 2px solid var(--foreground);
			transition: border-color 0.2s ease-in-out;
		}
	}
	${commonStyles}
	.DateInput {
		width: 100%;
	}
`

const DateRangeWrapper = styled.div`
	.DateRangePicker {
		height: 100%;
		& > div {
			height: 100%;
		}
		&_picker {
			background-color: var(--background);
		}
	}
	.DateRangePickerInput {
		height: 100%;
		background-color: transparent;
		&__withBorder {
			border: none;
			border-bottom: 2px solid var(--foreground);
			transition: border-color 0.2s ease-in-out;
		}
		&_arrow_svg,
		&_clearDates_svg {
			fill: var(--foreground);
			transition: fill 0.22s ease-in-out;
		}
	}

	${commonStyles}
`
