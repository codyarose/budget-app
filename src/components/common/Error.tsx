import React, { FC } from 'react'
import styled from 'styled-components'

interface Props {
	error: boolean
	helperText: string
}

const Error: FC<Props> = ({ error, helperText }) => {
	return <>{error && <StyledError>{helperText}</StyledError>}</>
}

export default Error

const StyledError = styled.div`
	font-style: italic;
`
