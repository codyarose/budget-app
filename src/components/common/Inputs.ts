import styled, { css } from 'styled-components'
import { transitions } from 'polished'
const baseStyles = css`
	background-color: transparent;
	color: ${({ theme }) => theme.colors.fg};
	font-size: 1rem;
	border: none;
	border-bottom: 2px solid currentColor;
	padding: ${({ theme }) => `${theme.spacing.sm}`};
	${transitions(['color', 'border-color'], '0.2s ease-in-out')}
`

export const TextInput = styled.input`
	${baseStyles}
	&::placeholder {
		color: ${({ theme }) => theme.colors.fg};
		transition: color 0.2s ease-in-out;
	}
`

export const SelectInput = styled.select`
	${baseStyles}
`
