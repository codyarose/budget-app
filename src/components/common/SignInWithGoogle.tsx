import React, { FC } from 'react'
import styled from 'styled-components'

interface Props {
	onClick: () => void
}

export const SignInWithGoogle: FC<Props> = ({ onClick }) => {
	return (
		<>
			<style>
				@import
				url(https://fonts.googleapis.com/css2?family=Roboto:wght@500&display=swap&text=Sign%20in%20with%20Google);
			</style>
			<Button onClick={onClick}>
				<ButtonContentWrapper>
					<ButtonIcon>
						<ButtonIconImage>
							<ButtonSVG
								version="1.1"
								xmlns="http://www.w3.org/2000/svg"
								width="18px"
								height="18px"
								viewBox="0 0 48 48"
								className="abcRioButtonSvg"
							>
								<g>
									<path
										fill="#EA4335"
										d="M24 9.5c3.54 0 6.71 1.22 9.21 3.6l6.85-6.85C35.9 2.38 30.47 0 24 0 14.62 0 6.51 5.38 2.56 13.22l7.98 6.19C12.43 13.72 17.74 9.5 24 9.5z"
									></path>
									<path
										fill="#4285F4"
										d="M46.98 24.55c0-1.57-.15-3.09-.38-4.55H24v9.02h12.94c-.58 2.96-2.26 5.48-4.78 7.18l7.73 6c4.51-4.18 7.09-10.36 7.09-17.65z"
									></path>
									<path
										fill="#FBBC05"
										d="M10.53 28.59c-.48-1.45-.76-2.99-.76-4.59s.27-3.14.76-4.59l-7.98-6.19C.92 16.46 0 20.12 0 24c0 3.88.92 7.54 2.56 10.78l7.97-6.19z"
									></path>
									<path
										fill="#34A853"
										d="M24 48c6.48 0 11.93-2.13 15.89-5.81l-7.73-6c-2.15 1.45-4.92 2.3-8.16 2.3-6.26 0-11.57-4.22-13.47-9.91l-7.98 6.19C6.51 42.62 14.62 48 24 48z"
									></path>
									<path fill="none" d="M0 0h48v48H0z"></path>
								</g>
							</ButtonSVG>
						</ButtonIconImage>
					</ButtonIcon>
					<ButtonContents>
						<span>Sign in with Google</span>
					</ButtonContents>
				</ButtonContentWrapper>
			</Button>
		</>
	)
}

const Button = styled.button`
	width: 100%;
	max-width: 240px;
	/* background-color: #4285f4; */
	background-color: #fff;
	/* color: #fff; */
	color: rgba(0, 0, 0, 0.6);
	border-radius: 1px;
	border: none;
	padding: 0;
	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.25);
	transition: background-color 0.218s, border-color 0.218s, box-shadow 0.218s;
	cursor: pointer;
	outline: none;
	overflow: hidden;
	position: relative;
	text-align: center;
	vertical-align: middle;
	white-space: nowrap;
	&:hover {
		box-shadow: 0 0 3px 3px rgba(66, 133, 244, 0.3);
	}
`
const ButtonContentWrapper = styled.div`
	border: 1px solid transparent;
	height: 100%;
	height: 100%;
`
const ButtonIcon = styled.div`
	padding: 15px;
	background-color: #fff;
	border-radius: 1px;
	float: left;
`
const ButtonIconImage = styled.div`
	width: 18px;
	height: 18px;
`
const ButtonSVG = styled.svg`
	display: block;
`
const ButtonContents = styled.span`
	font-size: 1rem;
	font-family: Roboto;
	line-height: 48px;
	font-weight: 500;
	letter-spacing: 0.21px;
	margin-left: 6px;
	margin-right: 6px;
	vertical-align: top;
`
