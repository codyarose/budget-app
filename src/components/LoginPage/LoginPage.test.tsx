import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import * as redux from 'react-redux'

import LoginPage from './index'
import { loginUser, selectIsAuthed } from '../../features/auth/authSlice'
import { SignInWithGoogle } from '../common/SignInWithGoogle'

jest.mock('react-redux', () => ({
	useDispatch: jest.fn(),
	useSelector: jest.fn((fn) => fn()),
}))
jest.mock('../../features/auth/authSlice')

const useDispatchSpy = jest.spyOn(redux, 'useDispatch')
const mockDispatchFn = jest.fn()
useDispatchSpy.mockReturnValue(mockDispatchFn)

let wrapper: ShallowWrapper, props: any
beforeEach(() => {
	props = {
		history: { push: jest.fn() },
	}
	wrapper = shallow(<LoginPage {...props} />)
})

describe('LoginPage', () => {
	it('should render correctly', () => {
		;(selectIsAuthed as jest.Mock).mockReturnValue(false)

		expect(wrapper).toMatchSnapshot()
	})

	it('should call loginUser on button click', () => {
		wrapper.find(SignInWithGoogle).simulate('click')

		expect(loginUser).toHaveBeenCalled()
	})
})
