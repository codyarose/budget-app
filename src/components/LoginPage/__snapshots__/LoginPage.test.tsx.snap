// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render correctly 1`] = `
<div>
  <h3>
    Login
  </h3>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
